packages/controller/src/ControlConnection.ts

```diff
@@ -694,55 +678,29 @@ export default class ControlConnection extends BaseConnection {
 
 	async handleRoleDeleteRequest(request: lib.RoleDeleteRequest) {
 		let id = request.id;
-		let role = this._controller.userManager.roles.get(id);
-		if (!role) {
-			throw new lib.RequestError(`Role with ID ${id} does not exist`);
-		}
```

Comment:
With this check being removed, `userPermissionsUpdated` will be called even if the user did not have the role. This could request in extra network traffic. `Set.delete` returns true if the value was in the set which can be used to reimplement this check.

Reply:
Good idea. I added the check back for both for the role and for only updating modified users.